src/pipboy.cc

```diff
@@ -836,939 +924,1014 @@ static void pipboyDrawText(const char* text, int flags, int color)
+            PIPBOY_WINDOW_WIDTH,
+            gPipboyWindowBuffer + PIPBOY_WINDOW_WIDTH * PIPBOY_WINDOW_CONTENT_VIEW_Y + PIPBOY_WINDOW_CONTENT_VIEW_X,
+            PIPBOY_WINDOW_WIDTH);
+         //pipboyWindowRenderQuestLocationList(a1);
+         pipboyWindowRenderHolodiskList(-1);
+         windowRefreshRect(gPipboyWindow, & gPipboyWindowContentRect);
+         inputPauseForTocks(200);
+         _stat_flag = 1;
+      } else {
+         if (gPipboyWindowHolodisksCount != 0 && gPipboyWindowHolodisksCount >= a1 && gPipboyMouseX > 429) {
+            soundPlayFile("ib1p1xx1");
+            _holodisk = 0;
+            int realIndex = (_view_page_holodisk * PIPBOY_STATUS_HOLODISK_LINES) + (a1); // Adjust a1 for pagination
+            int index = 0;
+            for (; index < gHolodisksCount; index += 1) {
+               HolodiskDescription * holodiskDescription = & (gHolodiskDescriptions[index]);
+               if (gGameGlobalVars[holodiskDescription -> gvar] > 0) {
+                  if (realIndex - 1 == _holodisk) { // use realIndex instead of a1
+                     break;
+                  }
+                  _holodisk += 1;
+               }
+            }
+            _holodisk = index;
 
-                    // Back
-                    const char* text1 = getmsg(&gPipboyMessageList, &gPipboyMessageListItem, 201);
-                    pipboyDrawText(text1, PIPBOY_TEXT_ALIGNMENT_LEFT_COLUMN_CENTER, _colorTable[992]);
+            blitBufferToBuffer(_pipboyFrmImages[PIPBOY_FRM_BACKGROUND].getData() + PIPBOY_WINDOW_WIDTH * PIPBOY_WINDOW_CONTENT_VIEW_Y + PIPBOY_WINDOW_CONTENT_VIEW_X,
+               PIPBOY_WINDOW_CONTENT_VIEW_WIDTH,
+               PIPBOY_WINDOW_CONTENT_VIEW_HEIGHT,
+               PIPBOY_WINDOW_WIDTH,
+               gPipboyWindowBuffer + PIPBOY_WINDOW_WIDTH * PIPBOY_WINDOW_CONTENT_VIEW_Y + PIPBOY_WINDOW_CONTENT_VIEW_X,
+               PIPBOY_WINDOW_WIDTH);
+
+            pipboyWindowRenderHolodiskList(a1);
+            pipboyWindowRenderQuestLocationList(-1);
+            windowRefreshRect(gPipboyWindow, & gPipboyWindowContentRect);
+            inputPauseForTocks(200);
+            pipboyWindowDestroyButtons();
+            pipboyRenderHolodiskText();
+            pipboyWindowCreateButtons(0, 0, true);
+            _holo_flag = 1;
+         }
+      }
+   }
 
-                    if (gPipboyLinesCount >= 0) {
-                        gPipboyCurrentLine = gPipboyLinesCount;
-                    }
+   if (_stat_flag == 0) { // holodisk handling
+      if (_holo_flag == 0 || a1 < 1025 || a1 > 1027) {
+         return;
+      }
 
-                    // Done
-                    const char* text2 = getmsg(&gPipboyMessageList, &gPipboyMessageListItem, 214);
-                    pipboyDrawText(text2, PIPBOY_TEXT_ALIGNMENT_RIGHT_COLUMN_CENTER, _colorTable[992]);
+      if ((gPipboyMouseX > 459 && a1 != 1027) || a1 == 1026) {
+         if (gPipboyHolodiskLastPage <= _view_page) {
+            if (a1 != 1026) {
```

Comment:
Would be nice to deobfuscate/decypher variable names for any code that's being rewritten.

Reply:
My 2c is that it's better to leave de-obfuscation as a separate change to make it easier to review PRs that actually change semantics vs. ones that are just variable renaming.